pepper-apis/dsm-core/src/test/java/org/broadinstitute/dsm/juniperkits/JuniperKitStatusTest.java

```diff
@@ -0,0 +1,188 @@
+    NonPepperKitCreationService nonPepperKitCreationService = new NonPepperKitCreationService();
+    DDPInstance ddpInstance;
+    EasyPostUtil mockEasyPostUtil = mock(EasyPostUtil.class);
+
+    @Before
+    public void setupJuniperBefore() {
+        JuniperSetupUtil.setupUpAJuniperInstance(instanceName, instanceGuid, "Juniper-Mock", "JuniperTestProject");
+        JuniperSetupUtil.loadDSMConfig();
+        ddpInstance = DDPInstance.getDDPInstanceWithRoleByStudyGuid(instanceGuid, "juniper_study");
+        when(mockEasyPostUtil.checkAddress(any(), anyString())).thenReturn(true);
+    }
+
+    @After
+    public void deleteJuniperInstance() {
+        JuniperSetupUtil.deleteJuniperTestStudies();
+    }
+
+    @Test
+    public void statusForJuniperStudy() {
+        String participantId = "TEST_PARTICIPANT";
+        int rand = new Random().nextInt() & Integer.MAX_VALUE;
+        String kitType = "SALIVA";
+
+        String json = "{ \"firstName\":\"P\","
+                + "\"lastName\":\"T\","
+                + "\"street1\":\"415 Main st\","
+                + "\"street2\":null,"
+                + "\"city\":\"Cambridge\","
+                + "\"state\":\"MA\","
+                + "\"postalCode\":\"02142\","
+                + "\"country\":\"USA\","
+                + "\"phoneNumber\":\" 111 - 222 - 3344\","
+                + "\"juniperKitId\":\"JuniperTestKitId_" + rand + "\","
+                + "\"juniperParticipantID\":\"" + participantId + rand + "\","
+                + "\"forceUpload\":false,"
+                + "\"skipAddressValidation\":false,"
+                + "\"juniperStudyID\":\"Juniper-mock-guid\"}";
+
+        JuniperKitRequest mockJuniperKit = new Gson().fromJson(json, JuniperKitRequest.class);
+        List<KitRequestShipping> oldkits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", kitType);
+
+        nonPepperKitCreationService.createNonPepperKit(mockJuniperKit, "SALIVA", mockEasyPostUtil, ddpInstance);
+        List<KitRequestShipping> newKits = KitRequestShipping.getKitRequestsByRealm(instanceName, "overview", "SALIVA");
+        Assert.assertEquals(newKits.size(), oldkits.size() + 1);
+
+        StatusKitResponse kitResponse = (StatusKitResponse) nonPepperStatusKitService.getKitsBasedOnStudyName(instanceGuid);
+        Assert.assertNotNull(kitResponse);
+        Assert.assertNotNull(kitResponse.getKits());
+        Assert.assertNotEquals(0, kitResponse.getKits().size());
+        Assert.assertNotNull(
+                kitResponse.getKits().stream().filter(kitStatus -> kitStatus.getJuniperKitId().equals("JuniperTestKitId_" + rand))
+                        .findAny());
+        for (NonPepperKitStatusDto nonPepperKitStatusDto : kitResponse.getKits()) {
+            JuniperSetupUtil.deleteJuniperKit(nonPepperKitStatusDto.getJuniperKitId());
+        }
+    }
+
+    @Test
+    public void statusByJuniperKitId() {
+        String participantId = "TEST_PARTICIPANT";
```

Comment:
Since this part is identical to the previous test it can be moved to the class level.